public/app/features/dashboard-scene/scene/DashboardScene.test.tsx

```diff
@@ -43,12 +49,54 @@ describe('DashboardScene', () => {
 
         expect(scene.state.isDirty).toBe(true);
 
-        // verify can discard change
         scene.onDiscard();
-
         const gridItem2 = sceneGraph.findObject(scene, (p) => p.state.key === 'griditem-1') as SceneGridItem;
         expect(gridItem2.state.x).toBe(0);
       });
+
+      it.each`
+        prop              | value
+        ${'title'}        | ${'new title'}
+        ${'description'}  | ${'new description'}
+        ${'tags'}         | ${['tag1', 'tag2']}
+        ${'editable'}     | ${false}
+        ${'graphTooltip'} | ${1}
+      `(
+        'A change to $prop should set isDirty true',
+        ({ prop, value }: { prop: keyof DashboardSceneState; value: any }) => {
+          scene.setState({ [prop]: value });
+
+          expect(scene.state.isDirty).toBe(true);
+
+          // TODO: Discard doesn't restore the previous state
+          // scene.onDiscard();
+          // expect(scene.state[prop]).toBe(prevState);
+        }
+      );
+
+      // TODO: Make the dashboard to restore the defaults on discard
+      it.skip('A change to refresh picker interval settings should set isDirty true', () => {
+        const refreshPicker = dashboardSceneGraph.getRefreshPicker(scene)!;
+        const prevState = refreshPicker.state.intervals;
+        refreshPicker.setState({ intervals: ['10s'] });
+
+        expect(scene.state.isDirty).toBe(true);
+
+        scene.onDiscard();
+        expect(refreshPicker.state.intervals).toEqual(prevState);
+      });
+
+      // TODO: Make the dashboard to restore the defaults on discard
+      it.skip('A change to time zone should set isDirty true', () => {
+        const timeRange = scene.state.$timeRange!;
+        const prevState = timeRange.state.timeZone;
+        timeRange.setState({ timeZone: 'UTC' });
+
+        expect(scene.state.isDirty).toBe(true);
+
+        scene.onDiscard();
+        expect(timeRange.state.timeZone).toBe(prevState);
+      });
```

Comment:
@torkelo, @ivanortegaalba I think for this to work we gonna have to have a mechanism on individual scene objects that are abstractions over some concepts (i.e. dash links) to indicate an object is "dirty". Similar to what we do with top level DashboardScene. 

A simpler idea I had is that we may have a dashboard DTO considered an initial state of the Dashboard scene, and on discard restore the state based on it. But this will require implementing some function similar to transformSaveModelToScene, that will restore states of particular objects.

WDYT? I think exploring the first approach when we mark individual objects `dirty` may be better long term (i.e. for implementing undo/redo if we ever land on sth like this)

Reply:
> WDYT? I think exploring the first approach when we mark individual objects dirty may be better long term (i.e. for implementing undo/redo if we ever land on sth like this)

Maybe, but it needs to bubble up to the dashboard.state.isDirty anyway (as it controls the save button)

Reply:
About the discard issue, I think the main problem is we don't clone deeper objects, only the root one. So the nested objects are modified. It would be better to restore the previous state based on serializing back the original loaded dashboard.

As a newbie, I have a slightly different opinion about managing dirty state:
* Is `dirty` state only needed by DashboardScene serialization. Not always is going to be needed for every scene use case. Why should we modify every single object in the tree to add a state-specific to the dashboard use case?
* I like the approach of util, being able to go through the dashboard and serialize the current state, choosing what to add there. This is actually what we have with `transformSceneToSaveModel`. Maybe we can make a dashboard to listen to the changes and decide if it is dirty or not. Until now, this is how it works.
* While using the API, I thought something like this would be amazing `something.setState({foo, markAsDirty})`. This is not the right firm, but it gives the ability to the modifier to mark the dashboard as dirty. We could do this by adding methods to the dashboard scene like: `dashboard.setPersistingState()`. 

Anyways, managing `dirty` state won't affect the discard behavior.

Reply:
@ivanortegaalba 

No this is not correct, when we clone we clone the full state tree. 

So there is no need discard logic on sub object level